addons/test_mail/tests/test_mail_activity.py

```diff
@@ -141,6 +141,23 @@ def _employee_crash(*args, **kwargs):
                     'test_mail.mail_act_test_todo',
                     user_id=self.user_admin.id)
 
+        test_activity.user_id = self.user_employee
+        test_activity.flush_recordset()
+
+        # user can read activities assigned to him even if he has no access to the document
+        with patch.object(MailTestActivity, 'check_access_rights', autospec=True, side_effect=_employee_crash):
+            searched_activity = self.env['mail.activity'].with_user(self.user_employee).search(
+                [('id', '=', test_activity.id)])
+            searched_activity.read(['summary'])
+
+        # user can read_group activities assigned to him even if he has no access to the document
+        with patch.object(MailTestActivity, 'check_access_rights', autospec=True, side_effect=_employee_crash):
+            self.env['mail.activity'].with_user(self.user_employee).read_group(
```

Comment:
Don't forget to check activity is effectively present in returned result (here it checks it does not crash, but it does not check the activity is effectively returned)

Reply:
Yes indeed, thanks